src/theory/ff/util.cpp

```diff
@@ -94,6 +94,30 @@ bool isFfFact(const Node& n)
              && n[0][0].getType().isFiniteField());
 }
 
+bool isFfLeaf(const Node& n, const FfSize& field)
+{
+  return n.getType().isFiniteField()
+         && !(n.getKind() == Kind::FINITE_FIELD_ADD
```

Comment:
This method has some relationship to `Theory::isLeafOf` see https://github.com/cvc5/cvc5/blob/main/src/theory/theory.h#L334.

It would be best if this didn't case split on each of the FF kinds, so that if/when a new FF kind is added, this method doesn't need to change.

Reply:
Good observation!